File: Pandacap.Data/Relationships/CommunityBookmark.fs
namespace Pandacap.Data

open System
open System.ComponentModel.DataAnnotations.Schema

/// An ActivityPub actor that represents a Lemmy community.
type CommunityBookmark() =
    inherit RemoteActorRelationship()

    [<NotMapped>]
    member this.Uri =
        new Uri(this.ActorId)

    [<NotMapped>]
    member this.Host =
        this.Uri.Host

    [<NotMapped>]
    member this.Name =
        this.PreferredUsername

File: Pandacap.Database/CommunityBookmark.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Pandacap.Database
{
    /// <summary>
    /// A bookmarked Lemmy community.
    /// (Pandacap is not designed to follow Lemmy communities directly, but it can interact with them through ActivityPub and through Lemmy's public API.)
    /// </summary>
    public class CommunityBookmark : RemoteActorRelationship
    {
        [NotMapped]
        public Uri Uri => new(ActorId);

        [NotMapped]
        public string Host => Uri.Host;

        [NotMapped]
        public string? Name => PreferredUsername;
    }
}
